casanovo/denovo/model_runner.py

```diff
@@ -59,20 +63,35 @@ def __init__(
         self.loaders = None
         self.writer = None
 
-        best_filename = "best"
+        filenames = ("{epoch}-{step}", "best")
         if output_rootname is not None:
-            best_filename = f"{output_rootname}.{best_filename}"
+            filenames = tuple(
+                f"{output_rootname}.{curr_name}" for curr_name in filenames
+            )
+        curr_filename, best_filename = filenames
+
+        if overwrite_ckpt_check and output_dir is not None:
+            patterns = [r"epoch=\d+\-step=\d+\.ckpt", r"best\.ckpt"]
+            if output_rootname is not None:
+                patterns = [
+                    re.escape(output_rootname + ".") + pattern
+                    for pattern in patterns
+                ]
+            utils.check_dir(output_dir, patterns)
 
         # Configure checkpoints.
         self.callbacks = [
             ModelCheckpoint(
-                dirpath=config.model_save_folder_path,
+                dirpath=output_dir,
```

Comment:
**todo:** `output_dir` could still be `None` here, in which case the previous file existing checks have not been done. However, this will use the default Lightning directory, and so files can still get overwritten. I think we need to explicitly set `output_dir` earlier in the code.